aicsimageio/tests/writers/test_ome_zarr_writer.py

```diff
@@ -160,3 +160,52 @@ def test_ome_zarr_writer_scaling(
         assert len(xforms) == 1
         assert xforms[0]["type"] == "scale"
         assert xforms[0]["scale"] == expected_read_scales[i]
+
+
+@array_constructor
+@pytest.mark.parametrize(
+    "write_shape, chunk_dims, num_levels, expected_read_shapes",
+    [
+        (
+            (2, 4, 8, 16, 32),
+            (1, 1, 2, 16, 16),
+            2,
+            [(2, 4, 8, 16, 32), (2, 4, 8, 8, 16), (2, 4, 8, 4, 8)],
+        ),
+        (
+            (16, 32),
+            (2, 4),
+            2,
+            [(16, 32), (8, 16), (4, 8)],
+        ),
+    ],
+)
+@pytest.mark.parametrize("filename", ["e.zarr"])
+def test_ome_zarr_writer_chunks(
+    array_constructor: Callable,
+    write_shape: Tuple[int, ...],
+    chunk_dims: Tuple[int, ...],
+    num_levels: int,
+    filename: str,
+    expected_read_shapes: List[Tuple[int, ...]],
+) -> None:
+    arr = array_constructor(write_shape, dtype=np.uint8)
+
+    # Construct save end point
+    save_uri = get_resource_write_full_path(filename, LOCAL)
+    # clear out anything left over
+    shutil.rmtree(save_uri, ignore_errors=True)
```

Comment:
Rather than manually do this you could use the `tempdir` fixture provided by `pytest`, creating your file within that directory will ensure it is automatically cleaned up by `pytest` each test invocation.

Reply:
resolved in [69eeaa9](https://github.com/AllenCellModeling/aicsimageio/pull/519/commits/69eeaa9588c9712c04a96732c84274fe2826913c)